lib/hackerone/client/report.rb

```diff
@@ -126,6 +147,87 @@ def suggest_bounty(message:, amount:, bonus_amount: nil)
         Activities.build(response_body)
       end
 
+      ## Idempotent: change the state of a report. See STATES for valid values.
+      #
+      # id: the ID of the report
+      # state: the state in which the report is to be put in
+      #
+      # returns an HackerOne::Client::Report object or raises an error if
+      # no report is found.
+      def state_change(state, message = nil)
+        raise ArgumentError, "state (#{state}) must be one of #{STATES}" unless STATES.include?(state)
+
+        body = {
+          type: "state-change",
+          attributes: {
+            state: state
+          }
+        }
+
+        if message
+          body[:attributes][:message] = message
+        elsif STATES_REQUIRING_STATE_CHANGE_MESSAGE.include?(state)
+          fail ArgumentError, "State #{state} requires a message. No message was supplied."
+        else
+          # message is in theory optional, but a value appears to be required.
+          body[:attributes][:message] = ""
+        end
+
+        make_post_request("reports/#{id}/state_changes", request_body: body)
+      end
+
+      ## Idempotent: Add a report reference to a project
+      #
+      # id: the ID of the report
+      # state: value for the reference (e.g. issue number or relative path to cross-repo issue)
+      #
+      # returns an HackerOne::Client::Report object or raises an error if
+      # no report is found.
+      def add_report_reference(reference)
+        body = {
+          type: "issue-tracker-reference-id",
+          attributes: {
+            reference: reference
+          }
+        }
+
+        response_json = make_post_request("reports/#{id}/issue_tracker_reference_id", request_body: body)
+        @report = response_json[:relationships][:report][:data]
+        self
+      end
+
+      ## Idempotent: add the issue reference and put the report into the "triage" state.
+      #
+      # id: the ID of the report
+      # state: value for the reference (e.g. issue number or relative path to cross-repo issue)
+      #
+      # returns an HackerOne::Client::Report object or raises an error if
+      # no report is found.
+      def triage(reference)
+        add_report_reference(reference)
+        state_change(:triaged)
+      end
+
+      # Add a comment to a report. By default, internal comments will be added.
+      #
+      # id: the ID of the report
+      # message: the content of the comment that will be created
+      # internal: "team only" comment (true, default) or "all participants"
+      def add_comment(message, internal: true)
```

Comment:
What about having this return a `HackerOne::Client::Activities::CommentAdded`?

Reply:
That's a good idea!